Awful.apk/src/main/assets/javascript/thread.js

```diff
@@ -186,8 +186,9 @@ function processPosts(scopeElement) {
 		highlightOwnQuotes(scopeElement);
 	}
 
+    // handle all GIFs that are not avatars
 	if (listener.getPreference('disableGifs') === 'true') {
-		scopeElement.querySelectorAll('img[title][src$=".gif"]').forEach(function each(gif) {
+		scopeElement.querySelectorAll('img[title][src$=".gif"]:not(.avatar)').forEach(function each(gif) {
```

Comment:
Since the `each(gif)` and the code at l216 are mostly the same, it would make sense to put it in a function, prepareFreezeGif or whatever.
Then just check if `this.naturalWidth === 1 && this.naturalHeight === 1 && this.dataset.avatarSecondSrc` and if so change the src. After that prepareFreezeGif if disableGifs is on.